Source/items.cpp

```diff
@@ -776,7 +812,80 @@ void CalcPlrItemVals(int p, BOOL Loadgfx)
 
 	if (plr[p]._pClass == PC_ROGUE) {
 		plr[p]._pDamageMod = plr[p]._pLevel * (plr[p]._pStrength + plr[p]._pDexterity) / 200;
-	} else {
+	}
+#ifdef HELLFIRE
+	else if (plr[p]._pClass == PC_MONK) {
+		if (plr[p].InvBody[INVLOC_HAND_LEFT]._itype != ITYPE_STAFF) {
+			if (plr[p].InvBody[INVLOC_HAND_RIGHT]._itype != ITYPE_STAFF && (plr[p].InvBody[INVLOC_HAND_LEFT]._itype != ITYPE_NONE || plr[p].InvBody[INVLOC_HAND_RIGHT]._itype != ITYPE_NONE)) {
+				plr[p]._pDamageMod = plr[p]._pLevel * (plr[p]._pStrength + plr[p]._pDexterity) / 300;
+			} else {
+				plr[p]._pDamageMod = plr[p]._pLevel * (plr[p]._pStrength + plr[p]._pDexterity) / 150;
+			}
+		} else {
+			plr[p]._pDamageMod = plr[p]._pLevel * (plr[p]._pStrength + plr[p]._pDexterity) / 150;
+		}
+	} else if (plr[p]._pClass == PC_BARD) {
+		if (plr[p].InvBody[INVLOC_HAND_LEFT]._itype == ITYPE_SWORD)
+			plr[p]._pDamageMod = plr[p]._pLevel * (plr[p]._pStrength + plr[p]._pDexterity) / 150;
+		else if (plr[p].InvBody[INVLOC_HAND_RIGHT]._itype == ITYPE_SWORD)
+			plr[p]._pDamageMod = plr[p]._pLevel * (plr[p]._pStrength + plr[p]._pDexterity) / 150;
+		else if (plr[p].InvBody[INVLOC_HAND_LEFT]._itype == ITYPE_BOW || plr[p].InvBody[INVLOC_HAND_RIGHT]._itype == ITYPE_BOW) {
+			plr[p]._pDamageMod = plr[p]._pLevel * (plr[p]._pStrength + plr[p]._pDexterity) / 250;
+		} else {
+			plr[p]._pDamageMod = plr[p]._pLevel * plr[p]._pStrength / 100;
+		}
+	} else if (plr[p]._pClass == PC_BARBARIAN) {
+
+		/*
+			if (v13 != 5)
+			goto LABEL_87;
+		v20 = v12->InvBody[4]._itype;
+		if (v20 == 2 || (v21 = v12->InvBody[5]._itype, v21 == 2) || v20 == 4 || v21 == 4) {

```

Comment:
Lets either finish this part or leave it out for now

Reply:
ye I'll finish